tests/infrastructure/test_kubernetes_job.py

```diff
@@ -158,7 +175,90 @@ async def test_task_group_start_returns_job_name(
         status_result = await tg.start(
             KubernetesJob(command=["echo", "hello"], name="test").run
         )
-        assert status_result == "mock-k8s-v1-job"
+        assert status_result == f"{FAKE_CLUSTER}:mock-k8s-v1-job"
+
+
+async def test_kill_calls_delete_namespaced_job(
+    mock_k8s_batch_client,
+    mock_k8s_client,
+    mock_watch,
+):
+    await KubernetesJob(command=["echo", "hello"], name="test").kill(
+        infrastructure_pid=f"{FAKE_CLUSTER}:mock-k8s-v1-job", grace_seconds=0
+    )
+
+    assert len(mock_k8s_batch_client.mock_calls) == 1
+    mock_call = mock_k8s_batch_client.mock_calls[0]
+    assert "call.delete_namespaced_job" in str(mock_call)
```

Comment:
Why not use `assert_called_once_with` here instead of asserting against the repr of the call?

Reply:
Fixed at https://github.com/PrefectHQ/prefect/pull/7701/commits/cdc067392171a4f64032f13049530ab195da01f2